examples/connected-car-cluster-sharding/akka-connected-car-streamlet/src/main/scala/connectedcar/streamlets/ConnectedCarCluster.scala

```diff
@@ -1,37 +1,59 @@
 package connectedcar.streamlets
 
-import akka.actor.{ ActorRef, Props }
-import akka.cluster.sharding.{ ClusterSharding, ClusterShardingSettings }
+import akka.actor.{ActorRef, Props}
+import akka.cluster.sharding.typed.scaladsl.{ClusterSharding, Entity, EntityRef, EntityTypeKey}
 import akka.util.Timeout
-import cloudflow.akkastream.scaladsl.{ FlowWithCommittableContext, RunnableGraphStreamletLogic }
+import akka.actor.typed.scaladsl.adapter._
+import akka.cluster.sharding.external.ExternalShardAllocationStrategy
+import akka.cluster.sharding.typed.ClusterShardingSettings
+import cloudflow.akkastream.scaladsl.{FlowWithCommittableContext, RunnableGraphStreamletLogic}
 import cloudflow.streamlets.StreamletShape
-import cloudflow.streamlets.avro.{ AvroInlet, AvroOutlet }
-import connectedcar.actors.ConnectedCarActor
+import cloudflow.streamlets.avro.{AvroInlet, AvroOutlet}
+import connectedcar.actors.{ConnectedCarActor, ConnectedCarERecordWrapper}
 
 import scala.concurrent.duration._
 import akka.pattern.ask
-import cloudflow.akkastream.{ AkkaStreamlet, Clustering }
-import connectedcar.data.{ ConnectedCarAgg, ConnectedCarERecord }
+import akka.stream.scaladsl.Sink
+import cloudflow.akkastream.{AkkaStreamlet, Clustering}
+import connectedcar.data.{ConnectedCarAgg, ConnectedCarERecord}
 
 object ConnectedCarCluster extends AkkaStreamlet with Clustering {
   val in    = AvroInlet[ConnectedCarERecord]("in")
   val out   = AvroOutlet[ConnectedCarAgg]("out", m ⇒ m.driver.toString)
   val shape = StreamletShape(in).withOutlets(out)
 
   override def createLogic = new RunnableGraphStreamletLogic() {
-    def runnableGraph = sourceWithCommittableContext(in).via(flow).to(committableSink(out))
+    val typeKey = EntityTypeKey[ConnectedCarERecordWrapper]("Car")
 
-    val carRegion: ActorRef = ClusterSharding(context.system).start(
-      typeName = "Counter",
-      entityProps = Props[ConnectedCarActor],
-      settings = ClusterShardingSettings(context.system),
-      extractEntityId = ConnectedCarActor.extractEntityId,
-      extractShardId = ConnectedCarActor.extractShardId
-    )
+    val source = shardedSourceWithCommittableContext(in,
+                                      typeKey: EntityTypeKey[ConnectedCarERecordWrapper],
+                                      (msg: ConnectedCarERecordWrapper) => msg.record.carId+""
+                                    )
+
+    val clusterSharding = ClusterSharding(system.toTyped)
+
+    def runnableGraph = source.map(kafkaEnvelope => {
+
+      clusterSharding.init(
+        Entity(typeKey)(createBehavior = _ => ConnectedCarActor())
+          .withAllocationStrategy(new ExternalShardAllocationStrategy(system, typeKey.name))
```

Comment:
These strategies, can they be defaulted in  shardedSourceWithCommittableContext, and optionally passed as arguments to it?